krm-functions/lib/interface/v1alpha1/interface_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+ Copyright 2023 Nephio.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+    "testing"
+
+    "github.com/google/go-cmp/cmp"
+    nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
+)
+
+var itface = `apiVersion: req.nephio.org/v1alpha1
+kind: Interface
```

Comment:
I am 90% sure that the way the code is written, we will lose comments in the YAML. Can you add a test here with comments, that goes through a read, edit, write cycle?

If I am right, then the pattern we can use instead os to:
 - on New, parse the thing into a KubeObject and cache that, it will be our internal storage format
 - GetInterface then would always then do KO -> YAML -> Interface, return that
 - Add a "SetInterface" that goes through all fields in the interface and does the corresponding KubeObject manipulations to set them in our internal storage
 
I suspect with `reflect` we could write helper libraries that do all the hard work *once* for all types.

Reply:
ok I changed the strategy accordingly